tools/src/icon4py/tools/py2fgen/wrappers/diffusion_wrapper.py

```diff
@@ -361,7 +390,7 @@ def grid_init_diffusion(
         num_edges=num_edges,
         vertical_size=vertical_size,
         limited_area=limited_area,
-        on_gpu=True if config_settings.device == "GPU" else False,
+        on_gpu=config_settings.device == settings.Device.GPU,
```

Comment:
string comparison resulted in always `False` which put some fields on host in gpu mode.

Reply:
Is not in ci therefore was forgotten in #575 .